tools/mason/MasonPublish.chpl

```diff
@@ -99,64 +93,136 @@ proc masonPublish(args: [] string) throws {
+    const name = getPackageName();
+    var safeDir = '';
+
+    if isLocal then safeDir = registryPath;
+    else {
+      safeDir = MASON_HOME + '/tmp/' + name + '-' + uniqueDir;
+    }
+
+    if !isLocal {
+      if !exists(MASON_HOME + '/tmp') then mkdir(MASON_HOME + '/tmp');
+      mkdir(safeDir);
+    }
+
+    if !isLocal {
+      cloneMasonReg(username, safeDir, registryPath);
+      branchMasonReg(username, name, safeDir, registryPath);
+    }
+
+    addPackageToBricks(packageLocation, safeDir, name, registryPath, isLocal);
+
+    if !isLocal {
+      gitC(safeDir + "/mason-registry", "git add .");
+      gitC(safeDir + "/mason-registry", "git commit -m '" + name + "'");
+      gitC(safeDir + "/mason-registry", 'git push --set-upstream origin ' + name, true);
+      rmTree(safeDir + '/');
+      writeln('--------------------------------------------------------------------');
+      writeln('Go to the above link to open up a Pull Request to the mason-registry');
+     }
+    else {
+      gitC(safeDir, 'git add Bricks/' + name);
+      gitC(safeDir, "git commit -m '" + name + "'" );
+    }
+
   }
-  catch {
-    if exists(safeDir) then rmTree(safeDir + '/');
-    writeln('Error publishing your package to the mason-registry');
+  catch e {
+    writeln(e.message());
   }
 }
 
 /* If --dry-run is passed then it takes the username and checks to see if the mason-registry is forked
  and the package has a git remote origin. If both exist then the package can be published.
  */
-proc dryRun(username: string) throws {
-  var fork = false;
-  var remoteCheck = checkIfForkExists(username: string);
-  if remoteCheck == 0 {
-    fork = true;
-  }
-  var git = false;
-  if doesGitOriginExist() {
-    git = true;
-  }
-  if git && fork {
-    writeln('Package can be published to the mason-registry');
-    writeln('Commands that will be run:');
-    writeln('> git clone git:github.com:[username]/mason-registry mason-registry');
-    writeln('> git checkout -b [package name]');
-    writeln('Package Name will be added to the Bricks in the mason-registry');
-    writeln('> git add .');
-    writeln('> git commit -m [package name]');
-    writeln('> git push --set-upstream origin [package name]');
+proc dryRun(username: string, registryPath : string, isLocal : bool) throws {
+  if !isLocal {
+    var fork = true;
+    var remoteCheck = checkIfForkExists(username: string);
+    if remoteCheck == 1 {
+      fork = false;
+    }
+    var git = false;
+    if doesGitOriginExist() {
+      git = true;
+    }
+    if git && fork {
+      writeln('Package can be published to the mason-registry');
+      writeln('Commands that will be run:');
+      writeln('> git clone git:github.com:[username]/mason-registry mason-registry');
+      writeln('> git checkout -b [package name]');
+      writeln('Package Name will be added to the Bricks in the mason-registry');
+      writeln('> git add .');
+      writeln('> git commit -m [package name]');
```

Comment:
With the commit msg changed above, maybe call this `[package info]` in `--dry-run`.